src/game/laramisc.c

```diff
@@ -573,7 +573,7 @@ void InitialiseLaraInventory(int32_t level_num)
 
 void LaraInitialiseMeshes(int32_t level_num)
 {
-    START_INFO *start = &g_GameInfo.start[level_num];
+    RESUME_INFO *start = &g_GameInfo.start[level_num];
```

Comment:
This needs sometimes to load from `current`, sometimes from `start` info.

For restarted levels, and levels started normally, it needs to load from `start`.
For loaded levels, it needs to load from `current`, otherwise it'd never load any pickups.

Make a new function for establishing the right `RESUME_INFO` for all three of the above Lara routines (untested):

```c
static RESUME_INFO *Lara_GetResumeInfo(void);

static RESUME_INFO *Lara_GetResumeInfo(int32_t level_num)
{
    if (g_GameInfo.level_type == GFL_SAVED) {
        return &g_GameInfo.current[level_num];
    }
    return &g_GameInfo.start[level_num];
}
```

and introduce `GF_RESTART_LEVEL`, `GFL_RESTARTED` and `GAME_INFO.level_type` I spoke about in my earlier post.